test/widgets/test_systray.py

```diff
@@ -91,3 +94,47 @@ def test_systray_reconfigure_screens(manager_nospawn, minimal_conf_noscreen, bac
     manager_nospawn.c.reconfigure_screens()
 
     assert manager_nospawn.c.bar["top"].info()["widgets"][0]["name"] == "systray"
+
+
+def test_systray_icons(manager_nospawn, minimal_conf_noscreen, backend_name):
+    """Check icons are placed correctly."""
+
+    @Retry(ignore_exceptions=(AssertionError))
+    def wait_for_icons():
+        _, icons = manager_nospawn.c.widget["systray"].eval("len(self.tray_icons)")
+        assert int(icons) == 2
+
+    if backend_name == "wayland":
+        pytest.skip("Skipping test on Wayland.")
+
+    for prog in ("volumeicon", "vlc"):
+        if shutil.which(prog) is None:
+            pytest.skip(f"{prog} must be installed. Skipping test.")
```

Comment:
I wonder if we can have some redundancy here - by checking that the test suite has access to, say, 2 of a possible 3 or 4 options, then we increase the number of machines that will test this locally.

Reply:
Yeah, that's true. I was focusing more on CI (hence the additions to `ci.yml`).

Ok - I'll add some more e.g. `nm-applet` is a likely candidate.

Reply:
Thinking about this some more, I'm not sure I like the idea of a test launching other processes. The most important thing is that this passes on CI. If it's skipped locally, I'm ok with that.

Reply:
Yeah, I think that's fair. It is a bit weird.